### ISSUE TITLE
The loss value of the validation set does not seem to decrease

### ISSUE THREAD

**Issue by qfwysw** [tom-roddick/oft#12]
Hi all:
       When i run the code, the loss value of the training set keeps decreasing.However,the loss value of the validation set does not seem to decrease. I want to know where i did wrong. Can anybody help me?
 
![image](https://user-images.githubusercontent.com/53482524/110318962-2f550100-8049-11eb-8566-cb0890658e9d.png)

**Comment by qfwysw** [comment]
Setting batchsize to 8 does not decrease the loss value

**Comment by WardZheng** [comment]
I have same problem with you. Did you solve it ?

**Comment by qfwysw** [comment]
> I have same problem with you. Did you solve it ?
No, I'm trying to solve it. However, It seems no progress. Can you leave your contact information so that we can discuss it？

**Comment by tom-roddick** [comment]
Hi @qfwysw, sorry that I haven't had the chance to look at this before now. I've uploaded a new commit 04f7383 which should resolve the issue, let me know if you run into any more problems.

**Comment by qfwysw** [comment]
> Hi @qfwysw, sorry that I haven't had the chance to look at this before now. I've uploaded a new commit [04f7383](https://github.com/tom-roddick/oft/commit/04f73835dc5be283eddc62e8dc31062b8bb9df5d) which should resolve the issue, let me know if you run into any more problems.

Thanks for your reply.I have done this but it can't work. However，I used heatmap loss to replace the Huber loss which worked well. The reason is that there is a huge imbalance between positive and negative  samples.  In your paper,  you define the negative samples which s < 0.05，but I can't find the definition in the heatmap loss.  I have tried to solve the problems above and start to? train，but l get the bad results. The Apple3d is 0.00, can you help me reproduce the paper？Thanks for you a lot，I have sent you email and look forward to your early reply.

**Comment by WardZheng** [comment]
> > I have same problem with you. Did you solve it ?
> > No, I'm trying to solve it. However, It seems no progress. Can you leave your contact information so that we can discuss it？

I think the network just overfit because the training dataset is too small. I use focal loss and new heads to train my own data, it seems work well.

**Comment by qfwysw** [comment]
> > > I have same problem with you. Did you solve it ?
> > > No, I'm trying to solve it. However, It seems no progress. Can you leave your contact information so that we can discuss it？
> 
> I think the network just overfit because the training dataset is too small. I use focal loss and new heads to train my own data, it seems work well.

Do you open the data augmentation? When I train the network without data augmentation, it works really well, but it overfits the training set. Yea, it seems that tom forget to balance the positive and negative samples. Can you share me your modification?

**Comment by WardZheng** [comment]
> > > > I have same problem with you. Did you solve it ?
> > > > No, I'm trying to solve it. However, It seems no progress. Can you leave your contact information so that we can discuss it？
> > 
> > 
> > I think the network just overfit because the training dataset is too small. I use focal loss and new heads to train my own data, it seems work well.
> 
> Do you open the data augmentation? When I train the network without data augmentation, it works really well, but it overfits the training set. Yea, it seems that tom forget to balance the positive and negative samples. Can you share me your modification?
Data set is the key! This is my email maysdluck@163.com

> > > > I have same problem with you. Did you solve it ?
> > > > No, I'm trying to solve it. However, It seems no progress. Can you leave your contact information so that we can discuss it？
> > 
> > 
> > I think the network just overfit because the training dataset is too small. I use focal loss and new heads to train my own data, it seems work well.
> 
> Do you open the data augmentation? When I train the network without data augmentation, it works really well, but it overfits the training set. Yea, it seems that tom forget to balance the positive and negative samples. Can you share me your modification?

Data set is the key! This is my email maysdluck@163.com

**Comment by qfwysw** [comment]
> > > > > I have same problem with you. Did you solve it ?
> > > > > No, I'm trying to solve it. However, It seems no progress. Can you leave your contact information so that we can discuss it？
> > > 
> > > 
> > > I think the network just overfit because the training dataset is too small. I use focal loss and new heads to train my own data, it seems work well.
> > 
> > 
> > Do you open the data augmentation? When I train the network without data augmentation, it works really well, but it overfits the training set. Yea, it seems that tom forget to balance the positive and negative samples. Can you share me your modification?
> > Data set is the key! This is my email [maysdluck@163.com](mailto:maysdluck@163.com)
> 
> > > > > I have same problem with you. Did you solve it ?
> > > > > No, I'm trying to solve it. However, It seems no progress. Can you leave your contact information so that we can discuss it？
> > > 
> > > 
> > > I think the network just overfit because the training dataset is too small. I use focal loss and new heads to train my own data, it seems work well.
> > 
> > 
> > Do you open the data augmentation? When I train the network without data augmentation, it works really well, but it overfits the training set. Yea, it seems that tom forget to balance the positive and negative samples. Can you share me your modification?
> 
> Data set is the key! This is my email [maysdluck@163.com](mailto:maysdluck@163.com)

Your email address may be wrong.

**Comment by qfwysw** [comment]
> I have same problem with you. Did you solve it ?

This is my email address fang.qu@foxmail.com

**Comment by IAMShashankk** [comment]
@qfwysw @WardZheng Could you please update your findings here? I am facing the same problem.
